src/core/path.c

```diff
@@ -832,6 +864,67 @@ static int path_can_start(Unit *u) {
         return 1;
 }
 
+static void activation_event_info_path_done(ActivationEventInfo *info) {
+        ActivationEventInfoPath *p = ACTIVATION_EVENT_INFO_PATH(info);
+
+        assert(p);
+
+        p->trigger_path_filename = mfree(p->trigger_path_filename);
+}
+
+static void activation_event_info_path_serialize(ActivationEventInfo *info, FILE *f) {
+        ActivationEventInfoPath *p = ACTIVATION_EVENT_INFO_PATH(info);
+
+        assert(info);
+        assert(f);
+        assert(p);
+
+        if (p->trigger_path_filename)
+                (void) serialize_item(f, "activation-event-info-path-filename", p->trigger_path_filename);
+}
+
+static int activation_event_info_path_deserialize(const char *key, const char *value, ActivationEventInfo **info) {
+        assert(key);
+        assert(value);
+
+        if (streq(key, "activation-event-info-path-filename")) {
+                if (!info || !*info)
+                        return -EINVAL;
+
+                ActivationEventInfoPath *p = ACTIVATION_EVENT_INFO_PATH(*info);
+                if (!p)
+                        return -EINVAL;
```

Comment:
to simplify things I'd really do this things outside of the streq() if block. I mean, `*info` must be a valid ActivationEventInfo object either way, so it makes sense to test this generically beforehand, instead of specifically for the `activation-event-info-path-filename` field.